src/app/shared/form/builder/ds-dynamic-form-ui/ds-dynamic-form-control-container.component.ts

```diff
@@ -150,32 +184,95 @@ export class DsDynamicFormControlContainerComponent extends DynamicFormControlCo
   @Input() hasErrorMessaging = false;
   @Input() layout = null as DynamicFormLayout;
   @Input() model: any;
+  relationships$: Observable<Array<SearchResult<Item>>>;
+  hasRelationLookup: boolean;
+  modalRef: NgbModalRef;
+  item: Item;
+  listId: string;
+  searchConfig: string;
+  selectedValues$: Observable<Array<{
+    selectedResult: SearchResult<Item>,
+    mdRep: MetadataRepresentation
+  }>>;
+  /**
+   * List of subscriptions to unsubscribe from
+   */
+  private subs: Subscription[] = [];
 
   /* tslint:disable:no-output-rename */
   @Output('dfBlur') blur: EventEmitter<DynamicFormControlEvent> = new EventEmitter<DynamicFormControlEvent>();
   @Output('dfChange') change: EventEmitter<DynamicFormControlEvent> = new EventEmitter<DynamicFormControlEvent>();
   @Output('dfFocus') focus: EventEmitter<DynamicFormControlEvent> = new EventEmitter<DynamicFormControlEvent>();
   @Output('ngbEvent') customEvent: EventEmitter<DynamicFormControlEvent> = new EventEmitter<DynamicFormControlEvent>();
   /* tslint:enable:no-output-rename */
-  @ViewChild('componentViewContainer', {read: ViewContainerRef}) componentViewContainerRef: ViewContainerRef;
+  @ViewChild('componentViewContainer', { read: ViewContainerRef }) componentViewContainerRef: ViewContainerRef;
 
   private showErrorMessagesPreviousStage: boolean;
 
   get componentType(): Type<DynamicFormControl> | null {
     return this.layoutService.getCustomComponentType(this.model) || dsDynamicFormControlMapFn(this.model);
   }
 
-  protected test: boolean;
   constructor(
     protected componentFactoryResolver: ComponentFactoryResolver,
     protected layoutService: DynamicFormLayoutService,
     protected validationService: DynamicFormValidationService,
-    protected translateService: TranslateService
+    protected translateService: TranslateService,
+    private modalService: NgbModal,
+    private relationService: RelationshipService,
+    private selectableListService: SelectableListService,
+    private itemService: ItemDataService,
+    private relationshipService: RelationshipService,
+    private zone: NgZone,
+    private store: Store<AppState>,
+    private submissionObjectService: SubmissionObjectDataService
   ) {
-
     super(componentFactoryResolver, layoutService, validationService);
   }
 
+  ngOnInit(): void {
```

Comment:
Could we describe this ngOnInit() better?  Either via TypeDocs or maybe inline comments.  It's pretty complex, and it'd be worth describing what it is doing when it finds a relationship.